tests/samples_test.py

```diff
@@ -0,0 +1,405 @@
+            numpy.sqrt(numpy.sqrt(2) - 1) / (2**0.75),
+            numpy.sqrt(numpy.sqrt(2) + 1) / (2**0.75),
+        ],
+    ),
+    (
+        smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 5),
+        [
+            0,
+            -1.0,
+            1.0,
+            -1 / numpy.sqrt(2),
+            1 / numpy.sqrt(2),
+            -numpy.sqrt(numpy.sqrt(2) + 1) / (2**0.75),
+            -numpy.sqrt(numpy.sqrt(2) - 1) / (2**0.75),
+            numpy.sqrt(numpy.sqrt(2) - 1) / (2**0.75),
+            numpy.sqrt(numpy.sqrt(2) + 1) / (2**0.75),
+            -numpy.sqrt(2 + numpy.sqrt(2 + numpy.sqrt(2))) / 2,
+            -numpy.sqrt(2 + numpy.sqrt(2 - numpy.sqrt(2))) / 2,
+            -numpy.sqrt(2 - numpy.sqrt(2 - numpy.sqrt(2))) / 2,
+            -numpy.sqrt(2 - numpy.sqrt(2 + numpy.sqrt(2))) / 2,
+            numpy.sqrt(2 - numpy.sqrt(2 + numpy.sqrt(2))) / 2,
+            numpy.sqrt(2 - numpy.sqrt(2 - numpy.sqrt(2))) / 2,
+            numpy.sqrt(2 + numpy.sqrt(2 - numpy.sqrt(2))) / 2,
+            numpy.sqrt(2 + numpy.sqrt(2 + numpy.sqrt(2))) / 2,
+        ],
+    ),
+    (smolyay.samples.NestedTrigonometricPointSet([0, 2 * numpy.pi], 0), [0]),
+    (
+        smolyay.samples.NestedTrigonometricPointSet([0, 2 * numpy.pi], 1),
+        [0, 2 * numpy.pi / 3, 4 * numpy.pi / 3],
+    ),
+    (
+        smolyay.samples.NestedTrigonometricPointSet([0, 2 * numpy.pi], 2),
+        2
+        * numpy.pi
+        * numpy.array(
+            [
+                0,
+                1 / 3,
+                2 / 3,
+                1 / 9,
+                2 / 9,
+                4 / 9,
+                5 / 9,
+                7 / 9,
+                8 / 9,
+            ]
+        ),
+    ),
+]
+
+sample_points_ids = [
+    "Non-nested Cheb [0]",
+    "Non-nested Cheb [1]",
+    "Non-nested Cheb [2]",
+    "Non-nested Cheb [3]",
+    "Non-nested Cheb [8]",
+    "Non-nested Trig [0]",
+    "Non-nested Trig [1]",
+    "Non-nested Trig [2]",
+    "Non-nested Trig [3]",
+    "Non-nested Trig [9]",
+    "Nested Cheb [0]",
+    "Nested Cheb [1]",
+    "Nested Cheb [2]",
+    "Nested Cheb [3]",
+    "Nested Slow Cheb [0]",
+    "Nested Slow Cheb [1]",
+    "Nested Slow Cheb [2]",
+    "Nested Slow Cheb [3]",
+    "Nested Slow Cheb [4]",
+    "Nested Slow Cheb [5]",
+    "Nested Trig [0]",
+    "Nested Trig [1]",
+    "Nested Trig [2]",
+]
+
+nested_sample_ids = [
+    "Nested Cheb [0]",
+    "Nested Cheb [1]",
+    "Nested Cheb [2]",
+    "Nested Cheb [3]",
+    "Nested Slow Cheb [0]",
+    "Nested Slow Cheb [1]",
+    "Nested Slow Cheb [2]",
+    "Nested Slow Cheb [3]",
+    "Nested Slow Cheb [4]",
+    "Nested Slow Cheb [5]",
+    "Nested Trig [0]",
+    "Nested Trig [1]",
+    "Nested Trig [2]",
+]
+
+
+def test_clenshaw_initial():
+    """test default properties"""
+    f = smolyay.samples.ClenshawCurtisPointSet([-2, 1], 7)
+    assert numpy.array_equal(f.domain, [-2, 1])
+    assert f.degree == 7
+    assert f._valid_cache == False
```

Comment:
I don't recommend including private components of the API in testing unless absolutely necessary. Better would be to have a test later where you get the points with some initial parameter value, change that value, and then check that the points update correctly
```suggestion
```